symengine/tests/cwrapper/test_cwrapper.c

```diff
@@ -63,8 +64,96 @@ void test_cwrapper() {
     basic_str_free(s);
 }
 
+void test_CVectorInt1()
+{
+    // Allocate on heap
+    CVectorInt *vec = vectorint_new();
+    vectorint_push_back(vec, 5);
+    assert(vectorint_get(vec, 0) == 5);
+    vectorint_free(vec);
+}
+
+struct X {
+    void *x;
+};
+
+void test_CVectorInt2()
+{
+    // Allocate on stack
+    CVectorInt *vec;
+
+    char data1[1];  // Not aligned properly
+    vec = (CVectorInt*)data1;
+    assert(vectorint_placement_new(vec, sizeof(data1)) == 2);
+
+    struct X data2[1];  // Aligned properly but small
+    vec = (CVectorInt*)data2;
+    assert(vectorint_placement_new(vec, sizeof(data2)) == 1);
+
+    char data3[50]; // Aligned properly and enough size to fit std::vector<int>
```

Comment:
That is correct. It's weird that it passes in Debug mode though...

Sent from my mobile phone.
On Jul 5, 2015 8:06 AM, "Isuru Fernando" <notifications@github.com> wrote:

> In symengine/tests/cwrapper/test_cwrapper.c
> <https://github.com/sympy/symengine/pull/495#discussion_r33895097>:
>
> > +};
> > +
> > +void test_CVectorInt2()
> > +{
> > +    // Allocate on stack
> > +    CVectorInt *vec;
> > +
> > +    char data1[1];  // Not aligned properly
> > +    vec = (CVectorInt*)data1;
> > +    assert(vectorint_placement_new(vec, sizeof(data1)) == 2);
> > +
> > +    struct X data2[1];  // Aligned properly but small
> > +    vec = (CVectorInt*)data2;
> > +    assert(vectorint_placement_new(vec, sizeof(data2)) == 1);
> > +
> > +    char data3[50]; // Aligned properly and enough size to fit std::vector<int>
>
> An array is only required to have the alignment of a member, which is 1
> byte in this case (alignment of char). For example,
>
>     char d[50];
>     std::cout << alignof(decltype(d)) << std::endl;
>
> gives me 1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/symengine/pull/495/files#r33895097>.
>